### ISSUE TITLE
[BUG] Set-PnPSite -NoScriptSite $false does not do anything

### ISSUE THREAD

**Issue by iloving** [pnp/powershell#2246]
### Notice

I have a script that makes the following call:

`Set-PnPSite -Identity $SiteFullURL -NoScriptSite $false`

Later in the script, I try to set a property bag value:
`Set-PnPPropertyBagValue -Key "$MYKEY" -Value "$MYVALUE" -Indexed`

But I get the following error:       | Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)) Site might have NoScript enabled, this prevents setting some property bag values.

### Expected behavior 
I expect it to do what it says in the documentation!

### Actual behavior 
Jack eff all

### Steps to reproduce behavior
What I listed above.  Create a site, enable scripting, try to set a property bag value.

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
1.11.46 nightly

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows
- [ ] Linux
- [ X] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by CallumCrowley** [comment]
@iloving Is the site "user-created"? You may need to enable a setting via the SharePoint admin center, before running the above PnP cmdlet:

[To allow custom script on OneDrive or user-created sites](https://docs.microsoft.com/sharepoint/allow-or-prevent-custom-script#to-allow-custom-script-on-onedrive-or-user-created-sites)

**Comment by iloving** [comment]
a) I'm not just some user.  I'm the global admin
b) That setting was enabled a long time ago, and things had been working fine until I updated my PnP module to the latest 1.11.0

**Comment by CallumCrowley** [comment]
@iloving when you say enabled, you mean you've already set it to "Allow users to run custom script on self-service created sites"?

**Comment by BaronSparky** [comment]
I did have problems with that argument, but I still use:
`Set-PnPTenantSite -Connection $newSiteConnection -Url {parameter:h_URL} -DenyAddAndCustomizePages:$true`
which works, even with 1.11.0

**Comment by iloving** [comment]
Here is what I am doing:

PS> New-PnPTenantSite -Url $SiteFullURL -Owner $SiteOwner -Title $SiteTitle -Template $SiteTemplate -TimeZone $TimeZone -RemoveDeletedSite -Wait #this works fine
PS> $Connection = Connect-PnPOnline -Url $SiteFullURL -ReturnConnection -Credentials $Credential

PS> Set-PnPTenantSite -Connection $Connection -Url $SiteFullURL -DenyAddAndCustomizePages:$false #This _used_ to work...
PS> Set-PnPPropertyBagValue -Key "$PropertyKey" -Value "$SiteType" -Indexed

Set-PnPPropertyBagValue: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)) Site might have NoScript enabled, this prevents setting some property bag values.
PS> Set-PnPSite -Identity $SiteFullURL -NoScriptSite:$false #this never worked
PS> Set-PnPPropertyBagValue -Key "$PropertyKey" -Value "$SiteType" -Indexed

Set-PnPPropertyBagValue: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)) Site might have NoScript enabled, this prevents setting some property bag values.

This used to work and now it doesn't work anymore.  I don't understand why.

**Comment by iloving** [comment]
Ok, I have confirmed that this functionality is indeed broken in 1.11.0.  I reverted back to 1.10.0 and my code works perfectly.

**Comment by gautamdsheth** [comment]
@iloving - can you please also pass the `-Connection` parameter to the failing cmdlet as well and check ? Or maybe remove `-ReturnConnection` parameter and then remove `-Connection` parameter from subsequent cmdlets and check ?

**Comment by iloving** [comment]
Okay, I've uninstalled 1.10.0 and reinstalled 1.11.0.  I don't know what changed from the last time I tried, but suddenly any cmdlet I used but forgot to explicitly provide the $Connection parameter, got very upset and threw an error about missing context.  (For the record, I am very happy about this.  Tracking connections through side-effect changes should never have been permitted in the first place, IMO) 

So I've gone through all the code and made sure every command has a -Connection explicitly specified... and now the Set-PnPTenantSite command is working even though I made no changes to it.  So....   🤷‍♀️
